### PR TITLE
[Feature](bangc-ops): add nms binary op

### PR DESCRIPTION
**Pull request by yangdian0315**
Thanks for your contribution and we appreciate it a lot. 

## 1. Motivation

nms二进制适配mluops

## 2. Modification

Please briefly describe what modification is made in this pull request, and indicate where to make the modification.

## 3. Test Report

If you want to know how to do operator testing, you can see [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).

#### 3.1.1 Accuracy Acceptance Standard

For static threshold standard details, see: [MLU-OPS Accuracy Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Accuracy-Acceptance-Standard.md).

- [ ] diff1: diff1 = 0
- [ ] diff2: diff2 = 0

#### 3.1.2 Operator Scheme checklist

|     No.        |                 Details              |            Check Results             |
|----------------|--------------------------------------|--------------------------------------|
|        1       |Supported hardware                    |             MLU370<br>MLU590         |
|        2       |Job types                             |          block <br> U1 <br> U4       |
|        3       |Layouts                               |          ARRAY etc      |
|        4       |Whether multi-dimensions are supported|                                      |
|        5       |Whether element zero is supported     |                                      |
|        6       |Data type(half/float)                 |           half / float etc           |
|        7       |Whether there is size limit           |                                      |

#### 3.1.3 New Feature Test

If you have checked the following items, please tick the relevant box.

- [ ] Data type test
- [ ] Multi-dimensional tensor test
- [ ] Layout test
- [ ] Different size/integer remainder end segment/alignment misalignment test
- [ ] Zero dimensional tensor test/zero element test
- [ ] stability test
- [ ] Multiple platform test
- [ ] Gen_case module test
- [ ] Nan/INF tests 
- [ ] Bug fix tests
- [ ] For memory leak check details, see[GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For code coverage check details, see: [GTest-User-Guide-zh](https://github.com/Cambricon/mlu-ops/blob/master/docs/GTest-User-Guide-zh.md).
- [ ] For I/O calculation efficiency check details, see: [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md).

#### 3.1.4 Parameter Check

When a new operator is submitted, the test points are given and the test results are stated.

|                   Test Point                    | Acceptance Standard | Test Result (Error Message) |
| ----------------------------------------------- | --------------------| --------------------------- |
| Whether it conforms to the operator restriction |     Normal error    |                             |
| Whether illegal parameters are passed           |     Normal error    |                             |

### 3.2 Accuracy Test

For the cases used in the New Feature Test section, the features and the number of cases are recorded here. When multiple operations are tested, multiple tables are needed to include details of these operations.

Operation:

|Test Point           | Description                      | Quantity |  Comment |
|----------           |----------------------------------|----------|  --------|
|Data type test       |half/float/i               |  Total 554 cases of 1 op(s).ALL PASSED       |          |
|Mult-tensor test     |Supports 1- dims              | Total 554 cases of 1 op(s) ALL PASSED       |          |
|Layout test          |Supports ARRAY                |          |          |
|Zero element test    |Whether to support this test      |          |          |
|Stability test       |--gtest_repeat=NUM<br>--thread=NUM|          |          |
|Mult-platform test   |MLU370/MLU590                     |          |          |
|Nan/INF test         |Whether to support this test      |          |          |
|Memory leak check    |Test result                       |          |          |
|Code coverage check  |Test result                       |          |          |

### 3.3 Performance Test

See [MLU-OPS Performance Acceptance Standard](https://github.com/Cambricon/mlu-ops/blob/master/docs/MLU-OPS-Performance-Acceptance-Standard.md) for details.

Platform：MLU370

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|-----|----|----|----|-----|
|op_name|    |    |     |    |    |    |     |
|op_name|    |    |     |    |    |    |     |

Platform：MLU590

|Operation|Mlu_hardware_time(us)|Mlu_interface_time(us)|Mlu_io_efficiency|Mlu_compute_efficiency|Mlu_workwpace_size(Bytes)|Data_type|Shape|
|-------|----|----|----|----|----|----|-----|
|op_name|    |    |    |    |    |    |     |
|op_name|    |    |    |    |    |    |     |

### 3.4 Summary Analysis

Please give a brief overview here, if you want to note and summarize the content.

### CODE REVIEW

**Review by GraceGuanH — COMMENTED** [review]
Please check the comments.

File: bangc-ops/mlu_op.h [hunk 1]
```diff
@@ -3474,6 +3551,306 @@ mluOpPolyNms(mluOpHandle_t handle,
...
+ * [4, boxes_num], [7, boxes_num] or [batches_num, 4, boxes_num].
+ * @param[in] pad_to_max_output_size
+ * When the \b pad_to_max_output_size set is true, the \b output will be padded to max_output_size
+ * with zero. Default is false.
+ * @par Returns
+ * - ::MLUOP_STATUS_SUCCESS, ::MLUOP_STATUS_BAD_PARAM
+ * *
+ * @par Scale Limitation
+ * - \b confidence_threshold should be in the range of [0, 1].
+ * - \b soft_nms_sigma should not be less than 0.0.
```
[the remark this answers is not in the document]
Comment (yangdian0315) [reply bangc-ops/mlu_op.h:3663]: 这里指的float的0  所以当时写的0.0

File: bangc-ops/mlu_op.h [hunk 2]
```diff
@@ -228,6 +228,72 @@ typedef enum {
...
+  MLUOP_NMS_BOX_CENTER   = 1,
+  /*!< The box mode is * [x_center, y_center, width, height] where width > 0 and * height > 0. */
+} mluOpNmsBoxPointMode_t;
+
+/*!
+ * @brief Describing the output modes that can be used to implement the nms operation.
+ */
+typedef enum {
+  MLUOP_NMS_OUTPUT_TARGET_INDICES = 0,
+  /*!< Returns target indices, sorted in decreasing order of confidences. */
```
[the remark this answers is not in the document]
Comment (yangdian0315) [reply bangc-ops/mlu_op.h:245]: 后面的confidences指 mluOpNms 接口里的输入

File: bangc-ops/mlu_op.h [hunk 3]
```diff
@@ -3474,6 +3551,304 @@ mluOpPolyNms(mluOpHandle_t handle,
...
+ * @par Example
+ * - None.
+ */
+mluOpStatus_t MLUOP_WIN_API
+mluOpCreateNmsDescriptor(mluOpNmsDescriptor_t *desc);
+
+// Group:Nms
+/*!
+ *
+ * @brief Destroys a Nms descriptor \b desc that was previously created with
```
Comment (GraceGuanH) [inline bangc-ops/mlu_op.h:3586]: an Nms

**Review by GraceGuanH — APPROVED** [review]

**Review by Unireverse — APPROVED** [review]

**Review by liuyuan1-v — APPROVED** [review]

**Review by cathyzhang222 — APPROVED** [review]

**Review by LingyanZhang-HAH — APPROVED** [review]

**Pull request merged**